tests/reset_workflow.go

```diff
@@ -203,190 +204,196 @@ func (s *FunctionalSuite) TestResetWorkflow() {
+		FunctionalSuite:     s,
+		tv:                  testvars.New(s.T().Name()),
+		reapplyExcludeTypes: []enumspb.ResetReapplyExcludeType{},
+		reapplyType:         enumspb.RESET_REAPPLY_TYPE_SIGNAL,
+	}
+	t.run()
 }
 
 func (s *FunctionalSuite) TestResetWorkflow_ExcludeNoneReapplyNone() {
-	s.testResetWorkflowReapply(
-		"exclude-none-reapply-none",
-		[]enumspb.ResetReapplyExcludeType{},
-		enumspb.RESET_REAPPLY_TYPE_NONE,
-	)
+	t := ResetTest{
+		FunctionalSuite:     s,
+		tv:                  testvars.New(s.T().Name()),
+		reapplyExcludeTypes: []enumspb.ResetReapplyExcludeType{},
+		reapplyType:         enumspb.RESET_REAPPLY_TYPE_NONE,
+	}
+	t.run()
 }
 
 func (s *FunctionalSuite) TestResetWorkflow_ExcludeSignalReapplyAll() {
-	s.testResetWorkflowReapply(
-		"exclude-signal-reapply-all",
-		[]enumspb.ResetReapplyExcludeType{enumspb.RESET_REAPPLY_EXCLUDE_TYPE_SIGNAL},
-		enumspb.RESET_REAPPLY_TYPE_ALL_ELIGIBLE,
-	)
+	t := ResetTest{
+		FunctionalSuite:     s,
+		tv:                  testvars.New(s.T().Name()),
+		reapplyExcludeTypes: []enumspb.ResetReapplyExcludeType{enumspb.RESET_REAPPLY_EXCLUDE_TYPE_SIGNAL},
+		reapplyType:         enumspb.RESET_REAPPLY_TYPE_ALL_ELIGIBLE,
+	}
+	t.run()
 }
 
 func (s *FunctionalSuite) TestResetWorkflow_ExcludeSignalReapplySignal() {
-	s.testResetWorkflowReapply(
-		"exclude-signal-reapply-signal",
-		[]enumspb.ResetReapplyExcludeType{enumspb.RESET_REAPPLY_EXCLUDE_TYPE_SIGNAL},
-		enumspb.RESET_REAPPLY_TYPE_SIGNAL,
-	)
+	t := ResetTest{
+		FunctionalSuite:     s,
+		tv:                  testvars.New(s.T().Name()),
+		reapplyExcludeTypes: []enumspb.ResetReapplyExcludeType{enumspb.RESET_REAPPLY_EXCLUDE_TYPE_SIGNAL},
+		reapplyType:         enumspb.RESET_REAPPLY_TYPE_SIGNAL,
+	}
+	t.run()
 }
 
 func (s *FunctionalSuite) TestResetWorkflow_ExcludeSignalReapplyNone() {
-	s.testResetWorkflowReapply(
-		"exclude-signal-reapply-none",
-		[]enumspb.ResetReapplyExcludeType{enumspb.RESET_REAPPLY_EXCLUDE_TYPE_SIGNAL},
-		enumspb.RESET_REAPPLY_TYPE_NONE,
-	)
+	t := ResetTest{
+		FunctionalSuite:     s,
+		tv:                  testvars.New(s.T().Name()),
+		reapplyExcludeTypes: []enumspb.ResetReapplyExcludeType{enumspb.RESET_REAPPLY_EXCLUDE_TYPE_SIGNAL},
+		reapplyType:         enumspb.RESET_REAPPLY_TYPE_NONE,
+	}
+	t.run()
 }
 
-func (s *FunctionalSuite) testResetWorkflowReapply(
-	testName string,
-	reapplyExcludeTypes []enumspb.ResetReapplyExcludeType,
-	reapplyType enumspb.ResetReapplyType,
-) {
-	totalSignals := 3
-	resetToEventID := int64(totalSignals + 1)
-
-	workflowID := fmt.Sprintf("functional-reset-workflow-test-%s", testName)
-	workflowTypeName := fmt.Sprintf("functional-reset-workflow-test-%s-type", testName)
-	taskQueueName := fmt.Sprintf("functional-reset-workflow-test-%s-taskqueue", testName)
-	identity := "worker1"
-
-	workflowType := &commonpb.WorkflowType{Name: workflowTypeName}
-	taskQueue := &taskqueuepb.TaskQueue{Name: taskQueueName, Kind: enumspb.TASK_QUEUE_KIND_NORMAL}
+type ResetTest struct {
```

Comment:
nit: this def can be private?

Reply:
Thanks for reviewing @yycptt. Made it private.